src/server.rs

```diff
@@ -69,6 +71,84 @@ fn replace_string_in_range(
     contents
 }
 
+fn function_defines(name: String, f: &FunctionExpr) -> bool {
+    for param in f.params.clone() {
+        if param.key.name == name {
+            return true;
+        }
+    }
+
+    false
+}
+
+fn is_scope(name: String, n: Rc<Node<'_>>) -> bool {
+    let mut dvisitor: DefinitionFinderVisitor =
+        DefinitionFinderVisitor::new(name);
+
+    walk::walk(&mut dvisitor, n.clone());
+
+    let state = dvisitor.state.borrow();
+
+    state.node.is_some()
+}
+
+fn find_references(
+    uri: lsp::Url,
+    contents: String,
+    position: lsp::Position,
+) -> Vec<lsp::Location> {
+    let mut locations: Vec<lsp::Location> = vec![];
+    let pkg = parse_and_analyze(&contents);
```

Comment:
I think I'd prefer the `parse_and_analyze` call to happen only in the `LspServer` implementation. The big reason for this is that I foresee at some point that the we keep a parsed/analyzed version in the store at some point, because we're doing these operations so often.